### PR TITLE
FABJ-539: loaded policy yaml file and close IO stream.

### PR DESCRIPTION
**Pull request by JcobCN**
Signed-off-by: 3htj <1786951843@qq.com>

### CODE REVIEW

File: src/main/java/org/hyperledger/fabric/sdk/ChaincodeEndorsementPolicy.java [hunk 1]
```diff
@@ -239,7 +239,9 @@ public void fromFile(File policyFile) throws IOException {
     @Deprecated
     public void fromYamlFile(File yamlPolicyFile) throws IOException, ChaincodeEndorsementPolicyParseException {
         final Yaml yaml = new Yaml(new SafeConstructor());
-        final Map<?, ?> load = (Map<?, ?>) yaml.load(new FileInputStream(yamlPolicyFile));
+        FileInputStream policyFileStream = new FileInputStream(yamlPolicyFile);
```
Comment (bestbeforetoday) [inline src/main/java/org/hyperledger/fabric/sdk/ChaincodeEndorsementPolicy.java:242]: Could you use try-with-resources to ensure the close is called?
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

File: src/main/java/org/hyperledger/fabric/sdk/ChaincodeEndorsementPolicy.java [hunk 2]
```diff
@@ -260,7 +262,9 @@ public void fromYamlFile(File yamlPolicyFile) throws IOException, ChaincodeEndor
 
     public static ChaincodeEndorsementPolicy fromYamlFile(Path yamlPolicyFile) throws IOException, ChaincodeEndorsementPolicyParseException {
         final Yaml yaml = new Yaml(new SafeConstructor());
-        final Map<?, ?> load = (Map<?, ?>) yaml.load(new FileInputStream(yamlPolicyFile.toFile()));
+        FileInputStream policyFileStream = new FileInputStream(yamlPolicyFile.toFile());
```
Comment (bestbeforetoday) [inline src/main/java/org/hyperledger/fabric/sdk/ChaincodeEndorsementPolicy.java:265]: As above, please use try-with-resources

**Review by bestbeforetoday — CHANGES_REQUESTED** [review]

**Comment by JiangCN** [comment]
The test can not past. But i did not modifiy there.

**Comment by bestbeforetoday** [comment]
A chaincode change highlighted a bad dependency in the test chaincode used by the SDK integration tests. I have a PR with a fix so I'll kick this build off again once that's in.

**Comment by bestbeforetoday** [comment]
The try-with-resources deals with closing the stream so the code does not need to do it explicitly. The try-block shouldn't be larger than necessary, so should just enclose the lines that use the stream, not all the code in the method.

To save time, I've made a similar change and removed the duplication of code between the two methods in PR #85 so this PR is no longer required.

Thank you for raising the issue and taking the time to submit a pull request with a candidate fix.

**Pull request closed without merging**